rl_projects/rl_ppaquette_project/supervised_model.py

```diff
@@ -0,0 +1,281 @@
+        self.power_embedding = nn.Embedding(
+            self.data_features["n_powers"], self.h_params["power_emb_size"]
+        )
+        uniform_(self.power_embedding.weight, -1, 1)
+
+        self.season_embedding = nn.Embedding(
+            self.data_features["n_seasons"], self.h_params["season_emb_size"]
+        )
+        uniform_(self.season_embedding.weight, -1, 1)
+
+        self.film_output_dims = [self.h_params["gcn_size"]] * (
+            self.h_params["n_graph_conv"] - 1
+        ) + [self.h_params["attn_size"] // 2]
+
+        # creating FiLM layer to hold weights for board encoding
+        self.board_encoding_film_weight_layer = nn.Linear(
+            in_features=self.h_params["power_emb_size"]
+            + self.h_params["season_emb_size"],
+            out_features=2 * sum(self.film_output_dims),
+            bias=True,
+        )
+        xavier_uniform_(self.board_encoding_film_weight_layer.weight)
+        zeros_(self.board_encoding_film_weight_layer.bias)
+
+        # creating FiLM layer to hold weights for previous orders
+        self.prev_orders_encoding_film_weight_layer = nn.Linear(
+            in_features=self.h_params["power_emb_size"]
+            + self.h_params["season_emb_size"],
+            out_features=2 * sum(self.film_output_dims),
+            bias=True,
+        )
+        xavier_uniform_(self.prev_orders_encoding_film_weight_layer.weight)
+        zeros_(self.prev_orders_encoding_film_weight_layer.bias)
+
+        # creating graph convolution layer for encoding board state
+        self.board_state_gcn_conv_layer = GraphConvolutionEncoder(
+            n_features=self.data_features["n_locations"],
+            input_size=self.data_features["n_location_features"],
+            output_gcn_size=self.h_params["gcn_size"],
+            final_output_gcn_size=self.h_params["attn_size"] // 2,
+            n_graph_conv_layers=self.h_params["n_graph_conv"],
+            batch_size=self.h_params["batch_size"],
+        )
+
+        # creating graph convolution layer for encoging previous orders
+        self.prev_orders_gcn_conv_layer = GraphConvolutionEncoder(
+            n_features=self.data_features["n_locations"],
+            input_size=self.data_features["n_orders_features"],
+            output_gcn_size=self.h_params["gcn_size"],
+            final_output_gcn_size=self.h_params["attn_size"] // 2,
+            n_graph_conv_layers=self.h_params["n_graph_conv"],
+            batch_size=self.h_params["batch_size"],
+        )
+
+        # Creating order embedding vector (to embed order_ix)
+        self.order_embedding = torch.empty(
+            (ORDER_VOCABULARY_SIZE, self.h_params["order_emb_size"])
+        ).uniform_(-1, 1)
+
+        # Creating candidate embedding
+        self.candidate_embedding = torch.empty(
+            (ORDER_VOCABULARY_SIZE, self.h_params["lstm_size"] + 1)
+        ).uniform_(-1, 1)
+
+    def _postprocess_input(self, inputs: dict):
```

Comment:
Is it meant to be "postprocessing", I see in the forward pass it the first thing that is done?